### ISSUE TITLE
[Feature request] Separate depth map input

### ISSUE THREAD

**Issue by ThreeDeeJay** [gkv311/sview#122]
As I mentioned in https://github.com/gkv311/sview/issues/117, 2D to 3D conversion has been improving a lot recently thanks to AI.
So I think it's just a matter of time until people start distributing depth maps of movies so that the player can take care of applying it onto the respective file to create the left and right views on-the-fly, as an alternative to the _less than legal_ method of distributing the whole movie in 3D (with the depth map pre-applied), which is what people have been doing, but it's also impractical in my opinion.

![image](https://github.com/gkv311/sview/assets/71472458/848dc05c-88fc-4d0f-8555-08566991fd2d)

Anyhow, this is just a suggestion to consider for the future in case user-made depth maps do become widespread (since the alternative of cooking a GPU/CPU for real-time conversion still seems farther into the future) and the demand for this feature increases. 🤔

**Comment by gkv311** [comment]
Depth map is a cheap way to generate multiple views from an image - I recall some vendors advertised it for their multi-view glasses-free displays.

But depth map + one color image pair is not enough for reconstructing a good-quality views. It will suffer from artifacts at large parallax and translucent surfaces. Something that 3d-reconstruction tools have fighted against for many years.

I've never been interested in this reconstruction/pseudostereo stuff. If somebody will make a research and will share some guidence on how to implement support of such images (with algorithms and file format description) - I may try implementing it myself, if it wouldn't be too complicated. Otherwise this ticket would wait for some developer interested in feature and sView (but most likely he will do that for some other project).

**Comment by ThreeDeeJay** [comment]
Agree, double camera/rendered views beats depth maps but at least a good one beats no depth, though those are hard to come by. The best free and user-friendly method I've found is DepthViewer, [here's a quick guide for more accurate results](https://github.com/parkchamchi/DepthViewer/issues/5#issuecomment-1657791622).

The [estimated](https://raw.githubusercontent.com/parkchamchi/DepthViewer/master/examples/demo_basic.png) depth maps ([PGM](https://netpbm.sourceforge.net/doc/pgm.html)) get cached in a .depthviewer (zip) file alongside metadata (which I just found out can be generated programmatically [like this](https://github.com/parkchamchi/DepthViewer/blob/master/DEPTH/depthpy/README.md)):
![demo_cache](https://raw.githubusercontent.com/parkchamchi/DepthViewer/master/examples/demo_cache.png)
![v7ms1d1lRx](https://github.com/gkv311/sview/assets/71472458/63335b20-dba7-4079-a4bd-f44beac0e038)

Example: 
![ImageGlass_ncIk0kHMbB](https://github.com/gkv311/sview/assets/71472458/4c85e523-b629-480e-abdf-9326ec7a264e)
[.depthviewer file](https://github.com/gkv311/sview/files/12935392/Interstellar_2014_Trailer_4_5.1-1080p-HDTN.mp4.dpt_beit_large_512.099cfd740503dade5c4cca0f225e6eba7ddb685bda5e7bcb977769792fdcd5c4.depthviewer.zip) - [2D source](http://videos.hd-trailers.net/Interstellar_2014_Trailer_4_5.1-1080p-HDTN.mp4)

I'm not sure if it would be more feasible to implement support for this particular format or read depth from a grayscale video (encoding PGMs, maybe saving filesize at the cost of some quality loss), but either way I'm guessing sView could just subdivide the virtual screen into a grid and use depth map pixel brightness to pull vertices towards the camera.

At this point, you're probably asking yourself: why not just use DepthViewer directly?
To be fair, while DV is fantastic for depth estimation and viewing that, it is fairly limited as a video player: it doesn't support as many viewing methods (just 2D, VR and coincidentally, 3D Vision) or projection (180/360). It also lacks surround/ambisonics/HRTF and a seek bar though that's on the roadmap as well as MKV container through the GUI 🤞

**Comment by ThreeDeeJay** [comment]
>  It will suffer from artifacts at large parallax and translucent surfaces.

I just found about [“2D plus DOT”](https://en.wikipedia.org/wiki/2D-plus-depth#Disadvantages) (Depth + Occlusion + Transparency)
> The critical limitations of 2D plus depth format are as-
sociated with occlusion and transparency rendering issues occur-
ring with multi-view video. An extension to 2D plus depth map
representation, referenced as “2D plus DOT” with additional
occlusion and transparency information allows to display higher
quality 3D multi-view video. Another disadvantage of 2D+depth
format is that depth map is difficult to create with accuracy, in
particular in real- time for 3D live events.
http://web.archive.org/web/20150525162501/http://enpub.fulton.asu.edu/resp/vpqm/vpqm10/Proceedings_VPQM2010/vpqm_p33.pdf#page=5

> An extension of the 2D plus depth approach is “2D plus DOT”; a 2D image plus depth, occlusion
and transparency data. This additional information adds support for carrying multiview video, i.e.
information on how the scene would look from a range of viewpoints, not just static left and right
eye views. The resulting multiview data set would, in principle, allow a viewer with a suitable
decoding and display device get more of a true 3D experience, where the scene would change
when the viewer moves his head from side to side to change his perspective.
> 
> The compression techniques required to support this data representation efficiently are still at an
early stage development. Assuming that they are successfully developed, the total bit-rate
required is likely to be about twice that of the bit-rate of a 2D signal.
However, data compression is only part of the problem. It is even more difficult to create
accurate depth, occlusion and transparency data in real-time than a simple depth map.
Furthermore, it is likely to be some time before a display device that is capable of supporting
more than stereoscopic viewing becomes available at consumer prices.
https://wikileaks.org/sony/docs/05/docs/DVB/Beyond_HDTV.PDF#page=28

Though I can't find any specific/technical info about it to see if it can be replicated, just mentions.
Who knows. Perhaps AI models will eventually become capable advanced enough to detect translucency.
